### DISCUSSION TITLE
Is there a way to create a custom straight edge?

### DISCUSSION THREAD

**Discussion by tompbsmith** [xyflow/xyflow#1875] [Q&A] [answered]
I've created a few custom edges that use `getSmoothStepPath` and `getBezierPath` but I want to create a custom edge that is just a straight edge but I'm not sure how to do it.

For example, in my custom smooth step edge I use:

`const edgePath = getSmoothStepPath({ sourceX, sourceY, sourcePosition, targetX, targetY, targetPosition, borderRadius: 0 });`

to get the edgePath veriable to pass to the `d` attribute of `<path />` but I'm not sure how to generate a straight path to pass to `d`.

Can someone help me out?

Sorry if this is very simple, I'm new to working with SVG data and so I've relied on `getSmoothStepPath` and `getBezierPath` so far.

Thanks

**Comment by moklick** [comment, marked as answer]
You can check the implementation of the `StraightEdge` component https://github.com/wbkd/react-flow/blob/main/src/components/Edges/StraightEdge.tsx#L45-L49

You can create a straight path by doing:

```js
const path = `M ${sourceX},${sourceY}L ${targetX},${targetY}`;
```

**Reply by tompbsmith** [reply]
Ah that's great, thank you for the help.
